### ISSUE TITLE
Cannot deploy microprofile jar to app service. Socket timeout and 404's

### ISSUE THREAD

**Issue by BenHunt-io** [microsoft/azure-maven-plugins#1221]
### Plugin name and version
**Name:** azure-webapp-maven-plugin
**Version:** 1.12
### Plugin configuration in your `pom.xml`

The formatting of the code seems to be off. It might easier to check at my repo: https://github.com/BenHunt-io/payaramicro-azure-helloworld.

`<build> 
    <finalName>payaramicro-hello-azure</finalName>  
    <plugins>
      <plugin>
        <groupId>com.microsoft.azure</groupId>
        <artifactId>azure-webapp-maven-plugin</artifactId>
        <version>1.12</version>
        <configuration>
          <schemaVersion>V2</schemaVersion>
          <resourceGroup>microprofile</resourceGroup>
          <appName>payaramicro-hello-azure-1610854104553</appName>
          <pricingTier>P1v2</pricingTier>
          <region>westeurope</region>
          <runtime>
            <os>linux</os>
            <javaVersion>java11</javaVersion>
            <webContainer>java11</webContainer>
          </runtime>
          <appSettings>
            <property>
              <name>PORT</name>
              <value>8080</value>
            </property>
          </appSettings>
          <deployment>
            <resources>
              <resource>
                <directory>${project.basedir}/target</directory>
                <includes>
                  <include>*.jar</include>
                </includes>
              </resource>
            </resources>
          </deployment>
        </configuration>
      </plugin>
    </plugins>
  </build>  `
### Expected behavior
I should be able to CURL https://payaramicro-hello-azure-1610854104553.azurewebsites.net/data/hello and get back "Hello World" after running `mvn azure-webapp:deploy`

### Actual behavior
`[INFO] Auth Type : AZURE_CLI, Auth Files : [/root/.azure/azureProfile.json, /root/.azure/accessTokens.json]
[INFO] [Correlation ID: 9f9a232e-0c13-4e05-b409-231beec5d41d] Instance discovery was successful
[INFO] Subscription : Azure subscription 1(9c5d1afe-64c0-451d-add4-c5fdea007a5b)
[INFO] Target Web App doesn't exist. Creating a new one...
[INFO] Creating App Service Plan 'ServicePlan2f24797f-7998-4de3'...
[INFO] Successfully created App Service Plan.
[INFO] Successfully created Web App.
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 2 resources to /home/ben/github/payaramicro-hello-azure/target/azure-webapp/payaramicro-hello-azure-1610854104553-4d831b6a-8ad6-4190-abfc-559f3f604d71
[INFO] Trying to deploy artifact to payaramicro-hello-azure-1610854104553...
[INFO] Renaming /home/ben/github/payaramicro-hello-azure/target/azure-webapp/payaramicro-hello-azure-1610854104553-4d831b6a-8ad6-4190-abfc-559f3f604d71/app-microbundle.jar to app.jar
[INFO] Deploying the zip package payaramicro-hello-azure-1610854104553-4d831b6a-8ad6-4190-abfc-559f3f604d714283029616987217113.zip...
[INFO] Exception occurred during deployment: java.net.SocketTimeoutException: timeout, retry immediately(1/3)...
[INFO] Successfully deployed the artifact to https://payaramicro-hello-azure-1610854104553.azurewebsites.net
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:42 min
[INFO] Finished at: 2021-01-16T23:19:49-06:00
[INFO] ------------------------------------------------------------------------`

### Steps to reproduce the problem
1. Clone my repo at https://github.com/BenHunt-io/payaramicro-azure-helloworld
2. Install azure-cli 2.17.0
3. Install the following Java version
`[root@bhuntblade15 payaramicro-hello-azure]$ java -version
openjdk version "11.0.9.1" 2020-11-04 LTS
OpenJDK Runtime Environment 20.11-(Zulu-11.43+55-linux_x64)-Microsoft-Azure-restricted (build 11.0.9.1+1-LTS)
OpenJDK 64-Bit Server VM 20.11-(Zulu-11.43+55-linux_x64)-Microsoft-Azure-restricted (build 11.0.9.1+1-LTS, mixed mode)`
4. Install the following Maven version:
`[root@bhuntblade15 payaramicro-hello-azure]$ mvn -v
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /mnt/c/Program Files/maven/apache-maven-3.6.3
Java version: 11.0.9.1, vendor: Azul Systems, Inc., runtime: /usr/lib/jvm/zulu-11-azure-amd64
Default locale: en, platform encoding: UTF-8
OS name: "linux", version: "5.4.72-microsoft-standard-wsl2", arch: "amd64", family: "unix"`
5. Run
`mvn clean package;
mvn azure-webapp:deploy`

- Make sure you have an azure subscription to allocate the resources under.

Other potentially meaningful notes:
- Building the app on Ubuntu-18.04 on WSL2

**Comment by Flanker32** [comment]
@BenHunt-io Thanks for your letter, i tried to clone your repository and run the jar locally with `java -jar payaramicro-hello-azure-microbundle.jar`, but it failed with "Failed to connect to 127.0.0.1 port 443: Connection refused", could you please help me check this issue?

**Comment by BenHunt-io** [comment]
I'm sorry you're not able to run locally. Do you have the correct java version installed and JAVA_HOME set for maven when you build the jar?

I'm essentially using this configuration off of [microprofile-starter.io](https://start.microprofile.io/)

![image](https://user-images.githubusercontent.com/25158608/104845044-04e09600-5899-11eb-8e26-5a5c205a6d5f.png)

**Comment by BenHunt-io** [comment]
I may have figured out the issue and it seems to be due to my ignorance on how micro profile and the payara-micro runtime actually work. It seems like the microbundle.jar depends on other things that are built from maven and can't be with with just the jar alone. I figured this out when I started making a docker container to deploy the application instead. I was only copying over the jar file and trying to run it. The server would start but no endpoints would be deployed.

The microprofile tutorial for azure https://docs.microsoft.com/en-us/azure/developer/java/eclipse-microprofile/deploy-microprofile-payara-micro-java-app-with-maven-plugin states to change the deployment from a war to a jar, but that is only for the plugin and not the maven packaging.

I'm pretty sure I came across another Microsoft tutorial where they switched the maven packaging from a war to a jar. It might have been for Spring boot or something of the like.
